DataFormats/Detectors/CTP/src/Scalers.cxx

```diff
@@ -702,8 +702,11 @@ std::pair<double, double> CTPRunScalers::getRateGivenT(double timestamp, int cla
   // orbit numbers
 
   // then we can use binary search to find the right entries
-  auto iter = std::lower_bound(mScalerRecordO2.begin(), mScalerRecordO2.end(), timestamp, [&](CTPScalerRecordO2 const& a, uint32_t value) { return a.epochTime < value; });
-  auto nextindex = iter - mScalerRecordO2.begin(); // this points to the first index that has orbit greater or equal to given orbit
+  auto iter = std::lower_bound(mScalerRecordO2.begin(), mScalerRecordO2.end(), timestamp, [&](CTPScalerRecordO2 const& a, double value) { return a.epochTime <= value; });
```

Comment:
@sawenzel @chiarazampolli  @noferini 
 It just occurred to me, there was previously `uint32_t` in the internal sample function of `getRateGivenT`.
That was implicitly casting the `double` timestamp to `uint32_t` which broke the comparison.
I corrected that. Apart from the `SOR` feature, that should solve some other issues we see.